ocaml/xapi/xapi_pool.ml

```diff
@@ -3747,6 +3747,23 @@ let set_local_auth_max_threads ~__context:_ ~self:_ ~value =
 let set_ext_auth_max_threads ~__context:_ ~self:_ ~value =
   Xapi_session.set_ext_auth_max_threads value
 
+let set_ext_auth_cache_enabled ~__context ~self ~value:enabled =
+  Db.Pool.set_ext_auth_cache_enabled ~__context ~self ~value:enabled ;
+  if not enabled then
+    Xapi_session.clear_external_auth_cache ()
+
+let set_ext_auth_cache_size ~__context ~self ~value:capacity =
+  if capacity < 0L then
+    raise Api_errors.(Server_error (invalid_cache_size, []))
+  else
+    Db.Pool.set_ext_auth_cache_size ~__context ~self ~value:capacity
```

Comment:
Don't you want to call `clear_external_auth_cache` here (and in the following function) to ensure that changes get picked up immediately, without having to restart xapi?

Reply:
OK, that's fine. If we were to actually publish this in the API docs, it would be good to mention this there (the need to disable+re-enable).